cmd/test.go

```diff
@@ -0,0 +1,290 @@
+		lines = append(lines, "  \\ \\  / /__ _ | | _   __| |  __ _ | |_  ___   _ __         ")
+		lines = append(lines, "   \\ \\/ // _` || || | / _` | / _` || __|/ _ \\ | '__|        ")
+		lines = append(lines, "    \\  /| (_| || || || (_| || (_| || |_| (_) || |           ")
+		lines = append(lines, "     \\/  \\__,_||_||_| \\__,_| \\__,_| \\__|\\___/ |_|           ")
+	default:
+		lines = append(lines, "                                                            ")
+		lines = append(lines, "                                                            ")
+		lines = append(lines, "                                                            ")
+		lines = append(lines, "                                                            ")
+		lines = append(lines, "                                                            ")
+		lines = append(lines, "                                                            ")
+	}
+
+	switch res.Score {
+	case categoryScoreA:
+		lines[0] += "          "
+		lines[1] += "    /\\    "
+		lines[2] += "   /  \\   "
+		lines[3] += "  / /\\ \\  "
+		lines[4] += " / ____ \\ "
+		lines[5] += "/_/    \\_\\"
+	case categoryScoreB:
+		lines[0] += " ____     "
+		lines[1] += "|  _ \\    "
+		lines[2] += "| |_) |   "
+		lines[3] += "|  _ <    "
+		lines[4] += "| |_) |   "
+		lines[5] += "|____/    "
+	case categoryScoreC:
+		lines[0] += "   ____     "
+		lines[1] += " / ____|   "
+		lines[2] += "| |       "
+		lines[3] += "| |       "
+		lines[4] += "| |____   "
+		lines[5] += " \\_____|  "
+	}
+
+	lines = append(lines, "")
+	lines = append(lines, fmt.Sprintf("%-60s%s", "TEST NAME", "RESULT"))
+	suggestions := []string{}
+	for name, singleTestRes := range res.TestsExecuted {
+		testOutput := ""
+		testOutput += fmt.Sprintf("%-60s", name)
+		if singleTestRes.Measurement != "" {
+			testOutput = strings.TrimSuffix(testOutput, strings.Repeat(" ", len(singleTestRes.Measurement)+1))
+			testOutput = testOutput + singleTestRes.Measurement + " "
+		}
+		testOutput += string(singleTestRes.Verdict)
+
+		if singleTestRes.Suggestion != "" {
+			suggestions = append(suggestions, singleTestRes.Suggestion)
+		}
+
+		if singleTestRes.Error != "" {
+			testOutput += " - " + singleTestRes.Error
+		}
+		lines = append(lines, testOutput)
+	}
+
+	if len(suggestions) != 0 {
+		lines = append(lines, "")
+		lines = append(lines, "IMPROVEMENT SUGGESTIONS")
+		lines = append(lines, suggestions...)
+	}
+
+	lines = append(lines, "")
+	lines = append(lines, res.ExecutionTime.String())
+
+	lines = append(lines, "")
+	for _, l := range lines {
+		_, err := w.Write([]byte(l + "\n"))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func calculateScore(results map[string]testResult) categoryScore {
+	// TODO(kalo): calculate score more elaborately (potentially use weights)
+	avg := 0
+	for _, t := range results {
+		if t.Verdict == testVerdictNotOk || t.Verdict == testVerdictBad || t.Verdict == testVerdictFail || t.Verdict == testVerdictTimeout {
```

Comment:
Why not switch/case construct? It would look more natural IMO

Reply:
Agree, first I had only 3 cases and if/else seemed better and forgot to change it later...